### DISCUSSION TITLE
Missing reference error (v1.3.2)

### DISCUSSION THREAD

**Discussion by CHENG-KH** [PGScatalog/pgsc_calc#192] [Q&A] [answered]
Hi, 
I have tried calculating the PGS score; however, there is an error. "ERROR nextflow.Channel - https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/raw/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar". Then, I went to the website, I found the correct pathway is "https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/tree/master/pgsc_calc/reference_data". How can I fix it?

Many thanks!

**Comment by nebfield** [comment, marked as answer]
It looks like you're trying to run an old version of the calculator. The latest version doesn't use that reference database. 

It's best to update `pgsc_calc` to version 2. A few things changed so it's a good idea to read the documentation after updating.

If you want to keep using version 1.3.2 you could fix this problem by changing the --ref parameter:

`--ref https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/raw/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar`

**Comment by CHENG-KH** [comment]
Thanks for your answer. 

If I want to update the PGSC-calc to version 2, how can I update it?

**Reply by nebfield** [reply]
```
$ nextflow pull pgscatalog/pgsc_calc
```

Should update the workflow, unless you've cloned the repository and run with `nextflow run main.nf ...`. In that case `git pull` on the main branch would work.

**Comment by CHENG-KH** [comment]
I have tried your suggestion, adding 
`--ref https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/blob/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar` into my command. However, the error still occurs as below. 

Core Nextflow options
  revision       : main
  runName        : voluminous_wing
  containerEngine: docker
  launchDir      : /NovaSeq_128/Kenny/PGSC_calc
  workDir        : /NovaSeq_128/Kenny/PGSC_calc/work
  projectDir     : /home/kenny77527/.nextflow/assets/pgscatalog/pgsc_calc
  userName       : kenny77527
  profile        : docker
  configFiles    : /home/kenny77527/.nextflow/assets/pgscatalog/pgsc_calc/nextflow.config

Input/output options
  input          : samplesheet.csv
  pgs_id         : PGS000004
  pgp_id         : null
  trait_efo      : null
  target_build   : GRCh38
  ref            : https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/blob/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar
  genotypes_cache: null
  outdir         : results

Variant Matching QC & Filtering
  liftover       : true

Institutional config options
  hostnames      : [:]

Max job request options
  max_cpus       : 2
  max_memory     : 16.GB

If you use pgscatalog/pgsc_calc for your analysis please cite:

* The Polygenic Score Catalog
  https://doi.org/10.1038/s41588-021-00783-5

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/pgscatalog/pgsc_calc/blob/master/CITATIONS.md
------------------------------------------------------
executor >  local (2)
[10/5c83c2] process > PGSCATALOG_PGSCALC:PGSCALC:DOWNLOAD_SCOREFILES ([pgs_id:PGS000004, pgp_id:, trait_efo:]) [  0%] 0 of 1
[38/2d647d] process > PGSCATALOG_PGSCALC:PGSCALC:INPUT_CHECK:SAMPLESHEET_JSON (samplesheet.csv)                [100%] 1 of 1 ✔
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:INPUT_CHECK:COMBINE_SCOREFILES                                -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_RELABELBIM                             -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_RELABELPVAR                            -
[skipped  ] process > PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF (Control chromosome ALL)           [100%] 1 of 1, stored: 1 ✔
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:MATCH_VARIANTS                                -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:MATCH_COMBINE                                 -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:APPLY_SCORE:PLINK2_SCORE                                      -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:APPLY_SCORE:SCORE_AGGREGATE                                   -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:APPLY_SCORE:SCORE_REPORT                                      -
[-        ] process > PGSCATALOG_PGSCALC:PGSCALC:DUMPSOFTWAREVERSIONS                                          -

ERROR ~ No such file or directory: https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/blob/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar

 -- Check '.nextflow.log' file for details
ERROR ~ ERROR: No scores calculated!

 -- Check '.nextflow.log' file for details
-[pgscatalog/pgsc_calc] Pipeline completed with errors-

**Reply by nebfield** [reply]
I've edited the URL, it should be https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/raw/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar

**Reply by CHENG-KH** [reply]
Thanks for your prompt response. The error turns into 
ERROR ~ No such file or directory: https://gitlab.ebi.ac.uk/nebfield/test-datasets/-/raw/master/pgsc_calc/reference_data/pgsc_calc_ref.sqlar

**Reply by nebfield** [reply]
Could you try downloading the file and setting a local path? Like `--ref ~/Downloads/pgsc_calc_ref.sqlar`

**Reply by CHENG-KH** [reply]
I have tried downloading the file, but it didn't work. 
So, I tried to update pgsc_calc to version 2.
I used what you recommended `nextflow pull pgscatalog/pgsc_calc`, and an error occurred. 

Checking pgscatalog/pgsc_calc ...
https://github.com/PGScatalog/pgsc_calc.git: connection failed

**Reply by nebfield** [reply]
That's a problem with your network (internet) connection. I'm not sure why you can't connect to GitHub. Perhaps you could talk to your system administrator if you're running the workflow on a HPC
